<ui version="4.0" >
 <class>TimeSignalWidget</class>
 <widget class="QWidget" name="TimeSignalWidget" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>301</width>
    <height>378</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Time signal generator</string>
  </property>
  <layout class="QVBoxLayout" >
   <property name="margin" >
    <number>9</number>
   </property>
   <property name="spacing" >
    <number>6</number>
   </property>
   <item>
    <layout class="QHBoxLayout" >
     <property name="margin" >
      <number>0</number>
     </property>
     <property name="spacing" >
      <number>6</number>
     </property>
     <item>
      <widget class="QPushButton" name="runStopButton" >
       <property name="font" >
        <font>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text" >
        <string>Run!</string>
       </property>
       <property name="checkable" >
        <bool>true</bool>
       </property>
       <property name="autoDefault" >
        <bool>true</bool>
       </property>
       <property name="default" >
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="stepButton" >
       <property name="text" >
        <string>Step</string>
       </property>
       <property name="flat" >
        <bool>false</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="resetButton" >
       <property name="text" >
        <string>Reset</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="speedFrame" >
     <property name="title" >
      <string>Speed</string>
     </property>
     <layout class="QGridLayout" >
      <property name="margin" >
       <number>9</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item row="1" column="1" >
       <widget class="QDoubleSpinBox" name="frequencyInput" >
        <property name="suffix" >
         <string> Hz</string>
        </property>
        <property name="decimals" >
         <number>6</number>
        </property>
        <property name="maximum" >
         <double>9999.989999999999782</double>
        </property>
        <property name="minimum" >
         <double>0.001000000000000</double>
        </property>
        <property name="value" >
         <double>10.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="1" >
       <widget class="QDoubleSpinBox" name="stepSizeInput" >
        <property name="suffix" >
         <string> s</string>
        </property>
        <property name="decimals" >
         <number>6</number>
        </property>
        <property name="maximum" >
         <double>999.990000000000009</double>
        </property>
        <property name="minimum" >
         <double>0.001000000000000</double>
        </property>
        <property name="value" >
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0" >
       <widget class="QLabel" name="frequencyLabel" >
        <property name="text" >
         <string>Frequency:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" >
       <widget class="QLabel" name="stepSizeLabel" >
        <property name="text" >
         <string>Step size:</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="signalFrame" >
     <property name="title" >
      <string>Time signal</string>
     </property>
     <layout class="QVBoxLayout" >
      <property name="margin" >
       <number>9</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item>
       <layout class="QHBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QLabel" name="startValueLabel" >
          <property name="text" >
           <string>Start value:</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="startValueInput" >
          <property name="decimals" >
           <number>6</number>
          </property>
          <property name="maximum" >
           <double>100000000.000000000000000</double>
          </property>
          <property name="minimum" >
           <double>0.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QRadioButton" name="compValueButton" >
        <property name="text" >
         <string>Computed value</string>
        </property>
        <property name="checked" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QLabel" name="stepLabel" >
          <property name="enabled" >
           <bool>true</bool>
          </property>
          <property name="sizePolicy" >
           <sizepolicy>
            <hsizetype>7</hsizetype>
            <vsizetype>5</vsizetype>
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text" >
           <string>Step:</string>
          </property>
          <property name="alignment" >
           <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="compValueStepInput" >
          <property name="enabled" >
           <bool>true</bool>
          </property>
          <property name="decimals" >
           <number>6</number>
          </property>
          <property name="maximum" >
           <double>100000000.000000000000000</double>
          </property>
          <property name="minimum" >
           <double>-100000000.000000000000000</double>
          </property>
          <property name="value" >
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="signalRatioLabel" >
          <property name="enabled" >
           <bool>true</bool>
          </property>
          <property name="sizePolicy" >
           <sizepolicy>
            <hsizetype>3</hsizetype>
            <vsizetype>5</vsizetype>
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="text" >
           <string>(10.0 x)</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QRadioButton" name="realTimeButton" >
        <property name="text" >
         <string>Realtime</string>
        </property>
        <property name="checked" >
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="signalFrame" >
     <property name="title" >
      <string>Signal</string>
     </property>
     <layout class="QHBoxLayout" >
      <property name="margin" >
       <number>9</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item>
       <widget class="QLabel" name="signalLabel" >
        <property name="sizePolicy" >
         <sizepolicy>
          <hsizetype>7</hsizetype>
          <vsizetype>5</vsizetype>
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text" >
         <string>--</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="signalShowButton" >
        <property name="text" >
         <string>show</string>
        </property>
        <property name="checked" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>compValueButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>stepLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>131</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel" >
     <x>91</x>
     <y>224</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>compValueButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>compValueStepInput</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>215</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel" >
     <x>166</x>
     <y>223</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>compValueButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>signalRatioLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>174</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel" >
     <x>244</x>
     <y>224</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>signalShowButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>signalLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>245</x>
     <y>307</y>
    </hint>
    <hint type="destinationlabel" >
     <x>79</x>
     <y>307</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runStopButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>speedFrame</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>75</x>
     <y>10</y>
    </hint>
    <hint type="destinationlabel" >
     <x>56</x>
     <y>89</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>runStopButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>signalFrame</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>86</x>
     <y>10</y>
    </hint>
    <hint type="destinationlabel" >
     <x>66</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>compValueButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>resetButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>235</x>
     <y>197</y>
    </hint>
    <hint type="destinationlabel" >
     <x>254</x>
     <y>11</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>compValueButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>stepButton</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>93</x>
     <y>182</y>
    </hint>
    <hint type="destinationlabel" >
     <x>124</x>
     <y>24</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
